### ISSUE TITLE
CustomNPC 1.18.2 create empty NBT tags

### ISSUE THREAD

**Issue by GodzzilaS** [magmamaintained/Magma-1.18.2#31]
**Minecraft Version:** 1.18.2

**Forge Version:** Client 40.2.0, Server 40.2.21

**Logs:** Check this [https://dropmefiles.com/jo3Fa](https://dropmefiles.com/jo3Fa)

**Steps to Reproduce:**
 1. Launch a game only with CustomNPC (latest snapshot)
 2. Try to stack items in inventory
 3. Try to start smelting items in furnace, open it, and close, and open one more time

**Description of issue:** CustomNPC creates a empty NBTtags for items, when you interacting with containers/items.
![image](https://github.com/user-attachments/assets/464f4fcc-9a8f-43df-bbe5-04ea11606342)

UPD any Additional Information:
[08:45:40] [Netty Server IO #2/ERROR] [no.mo.cr.CreateStuffAdditionsFix/]: Empty NBT tag was created for ItemStack: 2 porkchop
[08:45:40] [Netty Server IO #2/ERROR] [no.mo.cr.CreateStuffAdditionsFix/]: java.base/java.lang.Thread.getStackTrace(Thread.java:1619)
TRANSFORMER/minecraft@1.18.2/net.minecraft.world.item.ItemStack.handler$zza000$MagmaTestSetTag(ItemStack.java:1337)
TRANSFORMER/minecraft@1.18.2/net.minecraft.world.item.ItemStack.m_41751_(ItemStack.java)
TRANSFORMER/forge@40.2.21/net.minecraftforge.common.extensions.IForgeItem.readShareTag(IForgeItem.java:172)
TRANSFORMER/forge@40.2.21/net.minecraftforge.common.extensions.IForgeItemStack.readShareTag(IForgeItemStack.java:354)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.FriendlyByteBuf.m_130267_(FriendlyByteBuf.java:480)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.FriendlyByteBuf.m_178374_(FriendlyByteBuf.java:160)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.protocol.game.ServerboundContainerClickPacket.<init>(ServerboundContainerClickPacket.java:39)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.ConnectionProtocol$PacketSet.m_178327_(ConnectionProtocol.java:419)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.ConnectionProtocol.m_178321_(ConnectionProtocol.java:471)
TRANSFORMER/minecraft@1.18.2/net.minecraft.network.PacketDecoder.decode(PacketDecoder.java:32)

This is the output from the mod, which was supposed to track the appearance of empty NBT Tags. But this is didn't create any output when interact with furnace. There are a lot of places where NBT tags are used in the Custom NPC mod, as the source code showed. If necessary, I can share the source texts for any corrections

**Comment by TonimatasDEV** [comment]
CustomNPC is closed source. I need time for fix it.

**Comment by GodzzilaS** [comment]
Noppies gived a sources of this mod for me, i can give that to you. Because in this mod a lot of code and i found only 2-3 methods, I was able to fix only the bug in inventory, but not the furnaces, and it's so controversial

**Comment by TonimatasDEV** [comment]
If you can give me temporary reading access that would be great. Are you tried it in normal Forge and it works? (I think yes but to be sure)

**Comment by GodzzilaS** [comment]
No, i only tried on all cores (Mohist, CatServer, ArcLight). But i have this screenshot from his discord

Can i share git private repo to you?

![image](https://github.com/user-attachments/assets/578c413c-ff95-4426-b349-df233edcf56f)

**Comment by TonimatasDEV** [comment]
Yes, you can.

**Comment by TonimatasDEV** [comment]
It is only a problem with Furnace with your fix?

**Comment by GodzzilaS** [comment]
On the contrary, I could not fix the oven in any way

**Comment by GodzzilaS** [comment]
![image](https://github.com/user-attachments/assets/deabf617-5560-44a4-b59f-5fbff1702e9c)
This is about furnace.

Well, the smart developers of other mods said about my fix that it's a bad fix :D

**Comment by GodzzilaS** [comment]
![image](https://github.com/user-attachments/assets/d104dacc-def4-4e5f-bf56-9cb985d2ac82)
This is an assumption about why this bug basically happens.

**Comment by TonimatasDEV** [comment]
Can you send me the jar of the fixed version?

**Comment by GodzzilaS** [comment]
Where i can send this? Ds?

**Comment by GodzzilaS** [comment]
> > Where i can send this? Ds?
> 
> Yes, or comprime in a zip and send here.

Send in ds

**Comment by TonimatasDEV** [comment]
Sorry but I'm not going to maintain this anymore.
